### PR TITLE
Add multiselect mode state and checkbox

### PR DESCRIPTION
**Pull request by jonas-tai**
### Summary of changes

Add multiSelectMode and multiSelectSelectedAttributionIds to the state and a MultiSelect checkbox to the attributions view.

### Context and reason for change

Preparation for adding multi-select deletions.

### How can the changes be tested

Run tests or test checkbox with example file

### CODE REVIEW

File: src/Frontend/Components/AttributionList/AttributionList.tsx [hunk 1]
```diff
@@ -28,19 +36,22 @@ interface AttributionListProps {
   attributions: Attributions;
   selectedAttributionId: string | null;
   attributionIdMarkedForReplacement: string;
+
```
Comment (benedikt-richter) [inline src/Frontend/Components/AttributionList/AttributionList.tsx:39]: remove empty line

File: src/Frontend/state/actions/resource-actions/attribution-view-simple-actions.ts [hunk 2]
```diff
@@ -38,3 +42,21 @@ export function setAttributionIdMarkedForReplacement(
     payload: attributionIdMarkedForReplacement,
   };
 }
+
+export function setMultiSelectMode(
+  multiSelectMode: boolean
+): SetmultiSelectMode {
```
Comment (benedikt-richter) [inline src/Frontend/state/actions/resource-actions/attribution-view-simple-actions.ts:48]: -> `SetMultiSelectMode`

File: src/Frontend/state/actions/resource-actions/attribution-view-simple-actions.ts [hunk 3]
```diff
@@ -38,3 +42,21 @@ export function setAttributionIdMarkedForReplacement(
...
+  return {
+    type: ACTION_SET_MULTI_SELECT_MODE,
+    payload: multiSelectMode,
+  };
+}
+
+export function setmultiSelectSelectedAttributionIds(
```
Comment (benedikt-richter) [inline src/Frontend/state/actions/resource-actions/attribution-view-simple-actions.ts:55]: -> `setMultiSelectSelectedAttributionIds`

File: src/Frontend/state/actions/resource-actions/attribution-view-simple-actions.ts [hunk 4]
```diff
@@ -38,3 +42,21 @@ export function setAttributionIdMarkedForReplacement(
...
+  multiSelectSelectedAttributionIds: Array<string>
+): SetmultiSelectSelectedAttributionIds {
```
Comment (benedikt-richter) [inline src/Frontend/state/actions/resource-actions/attribution-view-simple-actions.ts:57]: -> `SetMultiSelectSelectedAttributionIds`

File: src/Frontend/Components/AttributionList/AttributionList.tsx [hunk 5]
```diff
@@ -87,11 +95,25 @@ export function AttributionList(props: AttributionListProps): ReactElement {
     );
   }
 
+  function handleMultiSelectModeChange(): (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => void {
+    return (event): void => {
+      dispatch(setMultiSelectMode(event.target.checked));
+    };
+  }
```
Comment (benedikt-richter) [inline src/Frontend/Components/AttributionList/AttributionList.tsx:104]: could now be
```
  function handleMultiSelectModeChange(
    event: React.ChangeEvent<HTMLInputElement>
  ) => void {
      dispatch(setMultiSelectMode(event.target.checked));
  }

```

File: src/Frontend/state/selectors/attribution-view-resource-selectors.ts [hunk 6]
```diff
@@ -25,3 +25,13 @@ export function getResourceIdsOfSelectedAttribution(
   }
   return [];
 }
+
+export function getMultiSelectMode(state: State): boolean {
+  return state.resourceState.attributionView.multiSelectMode;
+}
+
+export function getmultiSelectSelectedAttributionIds(
```
Comment (benedikt-richter) [inline src/Frontend/state/selectors/attribution-view-resource-selectors.ts:33]: -> getMultiSelectSelectedAttributionIds

**Review by benedikt-richter — APPROVED** [review]
just two small points left

**Review by benedikt-richter — APPROVED** [review]
nice :+1

**Review by leslielazzarino — APPROVED** [review]

**Pull request merged**